src/com/studentmanagement/dao/StudentDao.java

```diff
@@ -0,0 +1,207 @@
+        }
+        return studentList;
+    }
+
+    public boolean updateDob(Student student) throws SQLException{
+        String query = "UPDATE Student SET dob ='"+student.getDob()+"'"+ "WHERE ID="+student.getID();
+        Statement statement = getConnection().createStatement();
+        int resultSet = statement.executeUpdate(query);
+        if(resultSet != 0){
+            return true;
+        }
+        return false;
+    }
+    public boolean updateAllScore(double score) throws SQLException{
+        int length = findAll().size()+1;
+        String query = "UPDATE Student SET average_score ="+score+" WHERE ID BETWEEN 0 AND "+length;
+        Statement statement = getConnection().createStatement();
+        int resultSet = statement.executeUpdate(query);
+        if(resultSet !=0){
+            return true;
+        }
+        return false;
+    }
+    public int generateId() throws SQLException {
+        String query = "select max(id) as maxId from student ";
+        Statement stmt = getConnection().createStatement();
+        ResultSet resultSet = stmt.executeQuery(query);
+        if (resultSet.next()) {
+            return resultSet.getInt("maxId") + 1;
+        } else {
+            return 0;
+        }
+    }
+    public List<Student> searchByMajor(String major) throws SQLException{
+        String query = "SELECT * FROM Student WHERE major='"+major+"'";
+        Statement statement = getConnection().createStatement();
+        ResultSet resultSet = statement.executeQuery(query);
+        List<Student> studentList = new ArrayList<>();
+        if(resultSet.next()){
+            Student student = new Student();
+            studentList.add(student);
+        }
+        return studentList;
+    }
+
+    public boolean averageScoreStudent() throws  SQLException{
+        String query = "SELECT AVG(average_score) as AVERAGE from Student";
+        Statement statement = getConnection().createStatement();
+        ResultSet resultSet = statement.executeQuery(query);
+        if(resultSet.next()){
+            System.out.println("Average Score Is");
+            System.out.println(resultSet.getString(1));
+            return true;
+        }
+        return false;
+    }
+    public boolean modifyDataStudent(Student student) throws SQLException{
+       String query = "UPDATE Student SET last_name ='"+student.getLastName()+"', first_name ='"+student.getFirstName()+"',city ='"+student.getCity()+"', average_score ="+student.getAverageScore()+",gender='"+student.getGender()+
+               "', dob='"+student.getDob()+"', major='"+student.getMajor()+"'"+" WHERE ID="+student.getID(); ;
+        Statement statement = getConnection().createStatement();
+        int resultSet = statement.executeUpdate(query);
+        if(resultSet != 0){
+            return true;
+        }
+        return false;
+    }
+    public Student sortByScore() throws SQLException{
+        String query = "SELECT * FROM Student  ORDER BY average_score ASC";
```

Comment:
Ở đây e nên return student, sau đó in ra ở Service.
Công việc của dao chỉ là lấy data, sửa, xóa data
Việc hiển thị nằm ở các tầng bên trên: service và app.